tools/probes/probes_main.c

```diff
@@ -100,8 +101,17 @@ int get_buffer_file_free(struct wave_files *files)
 	return -1;
 }
 
+bool is_audio_format(uint32_t format)
+{
+	if ((format & PROBE_MASK_FMT_TYPE) && (format & PROBE_MASK_AUDIO_FMT) == 0)
+		return true;
```

Comment:
Is the above really correct? This is one of very rare cases where I get suspicious about missing parentheses - the above means `if ((format & PROBE_MASK_FMT_TYPE) != 0 && (format & PROBE_MASK_AUDIO_FMT) == 0)` - is that correct? If it is, then I'd use `== 0` and `!= 0` or just the boolean form consistently. And you can make it `return condition;`

Reply:
The logic is correct, but I can update to make it easier to follow.